pdata/plog/logs.go

```diff
@@ -12,66 +12,49 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package internal // import "go.opentelemetry.io/collector/pdata/internal"
+package plog // import "go.opentelemetry.io/collector/pdata/plog"
 
 import (
+	"go.opentelemetry.io/collector/pdata/internal"
 	otlpcollectorlog "go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/logs/v1"
 	otlplogs "go.opentelemetry.io/collector/pdata/internal/data/protogen/logs/v1"
 )
 
-// LogsToOtlp internal helper to convert Logs to otlp request representation.
-func LogsToOtlp(mw Logs) *otlpcollectorlog.ExportLogsServiceRequest {
-	return mw.orig
-}
-
-// LogsFromOtlp internal helper to convert otlp request representation to Logs.
-func LogsFromOtlp(orig *otlpcollectorlog.ExportLogsServiceRequest) Logs {
-	return Logs{orig: orig}
-}
-
-// LogsToProto internal helper to convert Logs to protobuf representation.
-func LogsToProto(l Logs) otlplogs.LogsData {
-	return otlplogs.LogsData{
-		ResourceLogs: l.orig.ResourceLogs,
-	}
-}
+// Logs is the top-level struct that is propagated through the logs pipeline.
+// Use NewLogs to create new instance, zero-initialized instance is not valid for use.
+type Logs internal.Logs
 
-// LogsFromProto internal helper to convert protobuf representation to Logs.
-func LogsFromProto(orig otlplogs.LogsData) Logs {
-	return Logs{orig: &otlpcollectorlog.ExportLogsServiceRequest{
-		ResourceLogs: orig.ResourceLogs,
-	}}
+func newLogs(orig *otlpcollectorlog.ExportLogsServiceRequest) Logs {
+	return Logs(internal.NewLogs(orig))
 }
 
-// Logs is the top-level struct that is propagated through the logs pipeline.
-// Use NewLogs to create new instance, zero-initialized instance is not valid for use.
-type Logs struct {
-	orig *otlpcollectorlog.ExportLogsServiceRequest
+func (ms Logs) getOrig() *otlpcollectorlog.ExportLogsServiceRequest {
+	return internal.GetOrigLogs(internal.Logs(ms))
 }
 
 // NewLogs creates a new Logs struct.
 func NewLogs() Logs {
-	return Logs{orig: &otlpcollectorlog.ExportLogsServiceRequest{}}
+	return newLogs(&otlpcollectorlog.ExportLogsServiceRequest{})
 }
 
 // MoveTo moves all properties from the current struct to dest
 // resetting the current instance to its zero value.
-func (ld Logs) MoveTo(dest Logs) {
-	*dest.orig = *ld.orig
-	*ld.orig = otlpcollectorlog.ExportLogsServiceRequest{}
+func (ms Logs) MoveTo(dest Logs) {
+	*dest.getOrig() = *ms.getOrig()
+	*ms.getOrig() = otlpcollectorlog.ExportLogsServiceRequest{}
 }
 
 // Clone returns a copy of Logs.
-func (ld Logs) Clone() Logs {
+func (ms Logs) Clone() Logs {
```

Comment:
Same Q here. Is it an accidental rename?

Reply:
Same answer.